scripts/run_e2e_tests_test.py

```diff
@@ -914,18 +869,28 @@ def mock_report_pass(unused_suite_name):
                 ],),
             ],
         )
-        report_swap = self.swap_with_checks(
+        report_flake_swap = self.swap_with_checks(
             flake_checker, 'report_pass', mock_report_pass,
             expected_args=[('full',)])
         exit_swap = self.swap_with_checks(
             sys, 'exit', mock_exit, expected_args=[(0,)])
-        with check_swap, setup_and_install_swap, register_swap, cleanup_swap:
-            with build_swap, start_webdriver_swap:
-                with start_google_app_engine_server_swap:
-                    with wait_swap, report_swap:
-                        with get_parameters_swap, popen_swap:
-                            with get_chrome_driver_version_swap, exit_swap:
-                                run_e2e_tests.main(args=[])
+
+        with contextlib2.ExitStack() as stack:
+            stack.enter_context(check_swap)
+            stack.enter_context(setup_and_install_swap)
+            stack.enter_context(register_swap)
+            stack.enter_context(cleanup_swap)
+            stack.enter_context(build_swap)
+            stack.enter_context(start_webdriver_swap)
+            stack.enter_context(dev_appserver_swap)
+            stack.enter_context(wait_swap)
+            stack.enter_context(report_flake_swap)
+            stack.enter_context(get_parameters_swap)
+            stack.enter_context(popen_swap)
+            stack.enter_context(exit_swap)
+            stack.enter_context(get_chrome_driver_version_swap)
```

Comment:
Could this be somehow simplified, maybe to just one function with multiple/tuple of args?

Reply:
ditto elsewhere

Reply:
Yep, better.